contentcuration/contentcuration/static/js/edit_channel/channel_list/mixins.js

```diff
@@ -0,0 +1,65 @@
+import _ from 'underscore';
+import { mapGetters, mapActions, mapMutations } from 'vuex';
+import { createTranslator } from 'utils/i18n';
+import { dialog, alert } from 'edit_channel/utils/dialog';
+
+const channelStrings = createTranslator('ChannelStrings', {
+  	unsavedChanges: "Unsaved Changes!",
+	unsavedChangesText: "Exiting now will undo any new changes. Are you sure you want to exit?",
+	dontSave: "Discard Changes",
+	keepOpen: "Keep Editing",
+	save: "Save",
+	errorChannelSave: "Error Saving Channel"
+});
+
+exports.setChannelMixin = {
+	computed: Object.assign(
+		mapGetters('channel_list', [
+			'activeChannel',
+			'changed'
+		]),
+		{
+			channelStrings() {
+				return channelStrings;
+			}
+		}
+	),
+	methods: Object.assign(
+		mapActions('channel_list', ['saveChannel']),
+		mapMutations('channel_list', {
+			setActiveChannel: 'SET_ACTIVE_CHANNEL',
+			cancelChanges: 'CANCEL_CHANNEL_CHANGES'
+		}),
+	    {
+			setChannel: function (channel) {
```

Comment:
Yes, the activeChannel gets used for the channel details panel

Reply:
Yeah, I assumed it was being used for something - was just wondering if it might be easier to pass around an id and store the id of the active channel, rather than the entire channel object - then grab the active channel from the shared channel data store.